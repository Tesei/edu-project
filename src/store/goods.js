import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import axios from 'axios'

export const useGoodsStore = defineStore(
    'goods',
    () => {
        const goods = ref([])
        const goodsList = computed(() => goods.value)
        const defaultGoods = ref([
            {
                id: 1,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара',
                description:
                    '3 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '10 000',
            },
            {
                id: 2,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 2',
                description:
                    '2 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '12 000',
            },
            {
                id: 3,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 3',
                description:
                    '1 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '13 000',
            },
            {
                id: 4,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 4',
                description:
                    '4 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '14 000',
            },
            {
                id: 5,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 5',
                description:
                    '4 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '14 000',
            },
            {
                id: 6,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 6',
                description:
                    '4 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '14 000',
            },
            {
                id: 7,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 7',
                description:
                    '4 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '14 000',
            },
            {
                id: 8,
                image: 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg',
                title: 'Наименование товара 7',
                description:
                    '4 Довольно-таки интересное описание товара в несколько строк. Довольно-таки интересное описание товара в несколько строк',
                price: '14 000',
            },
        ])
        const defaultImageLink = 'https://tesei.github.io/sites/mockUp\'s/polaroid.jpg'

        // Actions
        const fetchGoodsFromFakestore = async () => {
            try {
                let { data } = await axios.get('https://fakestoreapi.com/products', {
                    timeout: 5000,
                })
                //         {
                //     "id": 1,
                //     "title": "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
                //     "price": 109.95,
                //     "description": "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
                //     "category": "men's clothing",
                //     "image": "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
                //     "rating": {
                //         "rate": 3.9,
                //         "count": 120
                //     }
                // },
                goods.value = data
            } catch {
                console.warn('Устанавливаем массив дефолтных товаров')
                goods.value = defaultGoods.value
            }
        }

        function createPost(item) {
            goods.value.push(item)
        }
        function removePost(item) {
            goods.value = item.filter((p) => p.id !== goods.value.id)
        }

        return {
            fetchGoodsFromFakestore,
            defaultGoods,
            goods,
            defaultImageLink,
            createPost,
            removePost,
            goodsList,
            persist: true,
        }
    },
    { persist: true },
)
