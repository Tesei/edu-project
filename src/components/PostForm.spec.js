import { describe, it, expect, beforeAll, beforeEach } from 'vitest'
import { mount } from '@vue/test-utils'
import component from './PostForm.vue'
import { createTestingPinia } from '@pinia/testing'
import { useGoodsStore } from '@/store/goods.js'

import flushPromises from 'flush-promises'
import waitForExpect from 'wait-for-expect'

describe('PostForm component', () => {
    const buildWrapper = (stubActions = true) => {
        return mount(component, {
            global: {
                plugins: [createTestingPinia({ stubActions })],
            },
        })
    }

    it('mounts without error', () => {
        const wrapper = buildWrapper()
        expect(wrapper.exists).toBeTruthy()
    })
    it('Test the vee-validate check', async () => {
        const wrapper = buildWrapper()
        await wrapper.get('.aside__form').trigger('submit')
        await flushPromises()
        await waitForExpect(() => {
            expect(wrapper.html()).toContain('form__error')
        })
    })
    it('enter full data without input-errors', async () => {
        const wrapper = buildWrapper()
        await wrapper.find('[data-testid="title"]').find('input').setValue('TestEnter')
        await wrapper
            .find('[data-testid="image-link"]')
            .find('input')
            .setValue('https://tesei.github.io/sites/mockUp\'s/polaroid.jpg')
        await wrapper.find('[data-testid="price"]').find('input').setValue('10000')
        await wrapper.get('.aside__form').trigger('submit')
        await flushPromises()
        await waitForExpect(() => {            
            expect(wrapper.html()).not.toContain('form__error')
        })
    })
    it('added new goods for store', async () => {
        const wrapper = buildWrapper(false)
        const store = useGoodsStore()
        await wrapper.find('[data-testid="title"]').find('input').setValue('TestEnter')
        await wrapper
            .find('[data-testid="image-link"]')
            .find('input')
            .setValue('https://tesei.github.io/sites/mockUp\'s/polaroid.jpg')
        await wrapper.find('[data-testid="price"]').find('input').setValue('10000')
        await wrapper.get('.aside__form').trigger('submit')
        await flushPromises()        
        await waitForExpect(() => {
            expect(store.createPost).toHaveBeenCalledTimes(1)
        })
    })
})
