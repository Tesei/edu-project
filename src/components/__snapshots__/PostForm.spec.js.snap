// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostForm component > mounts without error 1`] = `
"<aside data-v-52374430="" class="main__aside aside">
  <div data-v-52374430="" class="aside__content">
    <form data-v-52374430="" class="aside__form form">
      <div data-v-52374430="" class="aside__forms">
        <div data-v-2f46d669="" data-v-52374430="" class="form__line aside__forms-item" data-testid="title"><label data-v-2f46d669="" for="0.9459949896618856" class="form__label form__label_req"> Наименование товара </label><input data-v-2f46d669="" type="text" class="input _req" id="0.9459949896618856" autocomplete="off" name="name" data-value="" placeholder="Введите наименование товара" data-error="">
          <!--v-if-->
        </div>
        <div data-v-35ce762b="" data-v-52374430="" class="form__line aside__forms-item"><label data-v-35ce762b="" for="0.0008887635054291021" class="form__label"> Описание товара </label><textarea data-v-35ce762b="" type="text" class="input" id="0.0008887635054291021" autocomplete="off" name="form[]" data-value="" placeholder="Введите описание товара" data-error="">        </textarea></div>
        <div data-v-2f46d669="" data-v-52374430="" class="form__line aside__forms-item" data-testid="image-link"><label data-v-2f46d669="" for="0.9141517989685859" class="form__label form__label_req"> Ссылка на изображение товара </label><input data-v-2f46d669="" type="text" class="input _req _active" id="0.9141517989685859" autocomplete="off" name="name" data-value="" placeholder="Введите ссылку" data-error="">
          <!--v-if-->
        </div>
        <!-- <my-checkbox
                        class="aside__forms-checkbox"
                        v-model="isStandartImage"
                        >
                        Применить стандартую картинку
                    </my-checkbox> -->
        <div data-v-2f46d669="" data-v-52374430="" class="form__line aside__forms-item" data-testid="price"><label data-v-2f46d669="" for="0.41301256582051527" class="form__label form__label_req"> Цена товара </label><input data-v-2f46d669="" type="text" class="input _req" id="0.41301256582051527" autocomplete="off" name="name" data-value="" placeholder="Введите цену" data-error="">
          <!--v-if-->
        </div>
      </div>
      <div data-v-43357113="" data-v-52374430="" class="button aside__button" type="submit"><button data-v-43357113="" type="submit" class="btn _active"> Добавить товар </button></div>
    </form>
  </div>
</aside>"
`;
